### PR TITLE
tiltfile: add os.realpath

### LINKED ISSUE #2098: Way to clean filepaths in Tiltfile
Requested by kavirajkanagaraj in #tilt:

> does anyone else miss helper function similar to Go’s filepath.Clean("//hello///how/are/you/") in Tiltfile? or python’s os.path.realpath()
> Its mainly combining different paths. We have multiple repositories for source code, helm charts and dockerfiles, so we just accept root of organisation and append different paths. The pain point is it may endup with multiple trailing slashes which causes problem particularly sending docker context. So good to have some helper function there!

It might be nice if there was some way to make it easier for external contributors to contribute to these functions.

See also #1980

### PR DESCRIPTION
**Pull request by jazzdan**
Fixes #2098

### CODE REVIEW

File: internal/tiltfile/os/os.go [hunk 1]
```diff
@@ -27,6 +28,11 @@ func (e Extension) OnStart(env *starkit.Environment) error {
 		return err
 	}
 
+	err = env.AddBuiltin("os.realpath", realpath)
```
Comment (nicks) [inline internal/tiltfile/os/os.go]: should this be os.path.realpath to mirror the python api?

File: internal/tiltfile/os/os.go [hunk 2]
```diff
@@ -61,3 +67,19 @@ func cwd(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs [
 	dir := starkit.AbsWorkingDir(t)
 	return starlark.String(dir), nil
 }
+
+func realpath(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var path starlark.String
```
Comment (nicks) [inline internal/tiltfile/os/os.go]: i would just make this a `string`, starlark will figure it out

File: internal/tiltfile/os/os.go (hunk 1 above)
Comment (jazzdan) [reply internal/tiltfile/os/os.go]: When I tried that I got an error saying "multi level modules are not currently supported".

Is that something we control or is that upstream?

Comment (nicks) [reply internal/tiltfile/os/os.go]: good question! lemme poke at it

**Pull request merged**